### PR TITLE
refactor(test): updates screen and scenario for test-tabs-tab-bar-layout-direction to include specific behaviors and edge-cases

### PR DESCRIPTION
**Pull request by LKuchno**
## Description

test-tabs-tab-bar-layout-direction files:

**scenario.md** has been updated to include specific behaviors and edge-cases:
- Android always override system settings with RN settings - note to execute scenario agains React Native settings were added
- iOS while picking inherit from list is setting value to system direction setting - when RN is set to LTR and system to RTL primary tab bar will be displayed in LTR but after reselect inherit value for TabHost it will be set to RTL (for RN: RTL and system:LTR start tab bar direction will be RTL and final: LTR)
- scenario were refactored to divide steps into System/RN rtl settings not TabHost 

**index.tsx** file was updated to fix the default value for the TabsHost layout direction. Now, displayed default value for TabsHost corresponds to the I18nManager.isRTL value, which is read by TabsContainer and applied to the direction prop based on that reading.

### CODE REVIEW

**Review by kkafar — APPROVED** [review]

File: apps/src/tests/single-feature-tests/tabs/test-tabs-tab-bar-layout-direction/scenario.md [hunk 1]
```diff
@@ -9,97 +9,179 @@ rendering by testing the precedence between system-level settings, React
 Native's I18nManager, and the explicit direction prop from
 react-native-screens.
 
-**OS test creation version:** iOS: 18.6 and 26.2, Android: API Level 36.
+On iOS, all scenarios follow a system direction × RN direction matrix,
+because `direction = inherit` falls back to the **native** app direction
+(i.e. the system setting), potentially overriding React Native's `forceRTL`.
+On Android, scenarios only set the RN direction, because `direction =
+inherit` propagates via `style.direction` through the view hierarchy and
+therefore follows the RN setting - no system-direction dimension is needed.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-tab-bar-layout-direction/scenario.md]: Let's maybe use something similar to description from the test app here?

File: apps/src/tests/single-feature-tests/tabs/test-tabs-tab-bar-layout-direction/scenario.md [hunk 2]
```diff
@@ -9,97 +9,179 @@ rendering by testing the precedence between system-level settings, React
...
+RTL language are NOT covered by E2E tests.
 
 ## Prerequisites
 
 - iOS device or simulator with at least one RTL language localization
-  configured in Xcode (e.g. empty ar.lproj/InfoPlist.strings), or system
-  language set to Arabic/Hebrew,
-- Android emulator with supportRtl enabled in app manifest.
+  configured in Xcode (e.g., an empty `ar.lproj/InfoPlist.strings`) and system
+  language set to Arabic/Hebrew.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/tabs/test-tabs-tab-bar-layout-direction/scenario.md]: I think we're mixing 2 things here:
1. Device/simulator system language setting
2. Application Xcode project's localizations.

```suggestion
- iOS device or simulator with system language set to Arabic/Hebrew.
- Valid RTL configuration in Xcode application project (RTL language set as app's default language or RTL language added to supported languages + at least one localization file, e.g. an empty `ar.lproj/InfoPlist.strings`)
```

**Review by kligarski — APPROVED** [review]
Looks good!
